### DISCUSSION TITLE
`timber/context` filter

### DISCUSSION THREAD

**Discussion by nlemoine** [timber/timber#2499] [General]
In 2.x, only global context is filterable. "post/posts" is not: https://github.com/timber/timber/blob/2.x/lib/Timber.php#L965-L995

Is there any reason that lead to that choice? I used to rename collections for better semantic in templates instead calling everything `post` or `posts` (e.g. `products`, `recipes`, etc.). Although this is a very opinionated approach, I think object/collections should still remain under the filter scope to allow user deal with these context keys the way they want.

**Comment by gchtr** [comment]
> Is there any reason that lead to that choice?

No, not really. We just went with what we already had (a filter for `timber/twig`) and expanded what Timber sets automatically in the context.

> Although this is a very opinionated approach, I think object/collections should still remain under the filter scope to allow user deal with these context keys the way they want.

I think this is not too opinionated and would go into the direction that Timber is taking in all sorts of places.

What kind of changes would you propose? Do you want to be able to

1. deactivate the default context and handle it yourself?
2. filter the names used for the variables that are set by default?

I think you mean 1), but just to be sure.

**Reply by nlemoine** [reply]
> I think this is not too opinionated and would go into the direction that Timber is taking in all sorts of places.

I'm absolutely not saying this should be the default behaviour 😅  I'm renaming context keys (`post`/`posts`) because having `posts` everywhere is confusing but this is my personal Timber setup. 

> What kind of changes would you propose?

Sorry, I mean 2) actually. I just think users should be able to choose how [`posts` and `post` are named](https://github.com/timber/timber/blob/2.x/lib/Timber.php#L965-L994) but they have been excluded from the `timber/context` filter in 2.x (that now is only applied in `context_global`)

**Reply by gchtr** [reply]
> I'm absolutely not saying this should be the default behaviour 😅

And I didn’t want to say it shouldn’t. What I wanted to say: We should provide a filter so that developers can choose to overwrite the default behavior.

What would the solution look like? Would you move the `timber/context` filter back to the `context()` function?

**Reply by nlemoine** [reply]
> What would the solution look like? Would you move the timber/context filter back to the context() function?

That would be fine! I have some few ideas for context handling. I'll come back when I get things more consolidated than it is right now. It has to do with #2412.
